### PR TITLE
wrap api client to add defaults

### PR DESCRIPTION
**Pull request by KPostOffice**
# Issue link
https://issues.redhat.com/browse/RHOAIENG-12802
# What changes have been made
If the user logged in using `oc login ...` the self signed CA cert wasn't being picked up

# Verification steps
Use `oc login` in a ODH cluster with self signed certs. The user should be able to create Ray Clusters without doing anything special

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: src/codeflare_sdk/cluster/auth.py [hunk 1]
```diff
@@ -211,6 +190,38 @@ def config_check() -> str:
...
+def get_api_client() -> client.ApiClient:
+    "This function should load the api client with defaults"
+    if api_client != None:
+        return api_client
+    to_return = client.ApiClient()
+    _client_with_cert(to_return)
+    return to_return
+
+
 def api_config_handler() -> Optional[client.ApiClient]:
```
Comment (ChristianZaccaria) [inline src/codeflare_sdk/cluster/auth.py:225]: Would this function become obsolete?

**Review by ChristianZaccaria — DISMISSED** [review]
Looks great! - Just a few fixes

**Review by Bobbins228 — COMMENTED** [review]
/lgtm

**Review by Bobbins228 — APPROVED** [review]
/lgtm /approve

**Review by ChristianZaccaria — APPROVED** [review]

**Pull request merged**